addons/point_of_sale/static/src/app/screens/partner_list/partner_list.xml

```diff
@@ -2,54 +2,59 @@
 <templates id="template" xml:space="preserve">
 
     <t t-name="point_of_sale.PartnerList">
-        <Dialog bodyClass="'partner-list overflow-y-auto'" contentClass="'h-100'">
+        <Dialog bodyClass="'partner-list overflow-y-auto position-relative'" contentClass="'h-100'">
```

Comment:
```suggestion
        <Dialog bodyClass="'partner-list overflow-y-auto position-relative'" contentClass="'h-100'">
      <div t-if="loadNewPartners.status === 'loading'" class="fs-1 h-100 w-100 bg-white position-absolute start-0 top-0 z-3 d-flex justify-content-center align-items-center">
                <i class="fa fa-spinner fa-spin" style="" />
            </div>

```

`useTrackedAsync` give us a `status` that will be loading during the request.

If you want more information about this method see here `odoo/addons/point_of_sale/static/src/app/utils/hooks.js`


Btw that's the bootstrap equivalent of your css code.